typing/env.mli

```diff
@@ -257,6 +257,7 @@ type lookup_error =
   | Non_value_used_in_object of Longident.t * type_expr * Jkind.Violation.t
   | No_unboxed_version of Longident.t
   | Error_from_persistent_env of Persistent_env.error
+  | Mutable_value_used_in_closure of string (* jra: Maybe rename this error/add other errors? *)
```

Comment:
Agreed - I think we'll want to generalize this now that we have more complicated locks - can discuss tomorrow.